ols/user_data_collection/data_collector.py

```diff
@@ -0,0 +1,245 @@
+"""Collect insights and upload it to the Ingress service.
+
+It collects the data on startup and then it waits for the specified interval.
```

Comment:
don't collect on startup.  have a 5 min delay before the first upload attempt at a minimum.  You do not want a case where the pod keeps crashing and then uploading data as soon as it starts, then crashes and uploads it again, rapidly.

Reply:
ok, I might need to have another env that disables this for tests though, like `RUN_WITHOUT_INITIAL_WAIT`.